libsrc/atari7800/exehdr.s

```diff
@@ -0,0 +1,46 @@
+;
+; Karri Kaksonen, 2022
+;
+; This header contains data for emulators
+;
```

Comment:
Please elaborate what this actually produces - IMHO the toolchain should produce binaries that run as is on actual hardware, not some kind of file with extra headers that only work in some emulated environment - producing those should be an extra step

Reply:
The 7800 has a lot of compile time configuration details. Without the header info the build is pretty useless. Headers are always generated on the Lynx target also by default. Having a header is the way a ROM burner knows how to burn the build to a working cart.

The a78 header appears to be the "standard" way to pass the information of the compilations. Therefore it is added to the default config file. By simply removing it from the config file you can generate just the binary.

What might be a good next step would be to have some defines in the config file that would automatically trigger the right bits in the header file.

Reply:
The header should perhaps be better a weak import, like in [atari-cart.cfg](https://github.com/cc65/cc65/blob/master/cfg/atari-cart.cfg) and [carthdr.s](https://github.com/cc65/cc65/blob/master/libsrc/atari/carthdr.s), then the user has full control about the process.

Any any case, what you say should be stated in the .cfg file and the documentation

Reply:
I don't really get it. I tried to use weak definition for EXEHDR and even tried to see what happens if I add CARTSIZE to trigger the definition of EXEHDR. Both experiments just add complexity for the user. The definition EXEHDR: type = import; is so clear. You want to import an EXEHDR.

If you have some insight in why you want to use more complicated techniques for triggering the inclusion of EXEHDR please educate me.

Reply:
using weak makes it optional, you can produce a binary with or without the extra header just by linking it or not - same as with the atari cartridge

Reply:
I believe the reason is because of the use cases. An ordinary user wants a workflow that is simple and helps to streamline the development. Picking the choices in the cfg file for many details is a good way to customize the build. Reflecting the choices in the header is a good way for documenting them.

But it is of course up to you what you want to have in your repository.

I pushed the changes I want to have in there.

Reply:
> Picking the choices in the cfg file for many details is a good way to customize the build.

However, then you will have to change the default cfg file. For cc65 the general rule always has been: once you do this, you are entirely on your own, and building the supported "normal" files will never require this. Ideally you shouldnt have to do this at all for common usecases.

In this case it seems like the way to go is something like:

- exehdr in a seperate file which is imported "weak" by the startup code
- exehdr itself imports a bunch of symbols "weak" which allow to configure the header

now the user can forceimport the header in his code (or just put it on the linker cmdline), and define the symbols he wants to change to configure it - no changes in the provided library or cfg file needed.

Reply:
I pushed the changes a minute ago.